### PR TITLE
Update release notes with RC1 and RC2 info

### PR DESCRIPTION
**Pull request by mfreed**
Update release notes with information about RC1 and RC2, as well as update about 2.0 description.

### CODE REVIEW

File: release-notes.md [hunk 1]
```diff
@@ -12,27 +12,128 @@ can view active developments on GitHub at any time.
...
+multiple instances of TimescaleDB, configured so that one instance serves
+as an access node and multiple others as data nodes. All queries for a
+distributed hypertable are issued to the access node, but inserted data
+and queries are pushed down across data nodes for greater scale and
+performance.
+
+This release also adds support for user-defined actions allowing users to
+define actions that are run by the TimescaleDB automation framework.
+
+In addition to these major new features, the 2.0 branch introduces _breaking_ changes
```
Comment (solugebefola) [inline release-notes.md:95]: I think the fact that there are breaking changes should be more emphasized, maybe with a minor headline?

File: release-notes.md [hunk 2]
```diff
@@ -12,27 +12,128 @@ can view active developments on GitHub at any time.
...
+to APIs and existing features, such as continuous aggregates. These changes are not
+backwards compatible and might require changes to clients and/or scripts that rely on
```
Comment (solugebefola) [inline release-notes.md:97]:
```suggestion
backwards compatible and may require changes to clients and/or scripts that rely on
```

**Review by solugebefola — APPROVED** [review]

File: release-notes.md (hunk 2 above)
Comment (mfreed) [reply release-notes.md:97]: Given that this is already in github history, and we typically just copy github notes into the docs, I'm going to skip this grammatical change.  Thanks for careful read, though.

File: release-notes.md (hunk 1 above)
Comment (mfreed) [reply release-notes.md:95]: Yeah, will add to front-piece, and we'll be sure to include in 2.0 announcement.

Comment (mfreed) [reply release-notes.md:95]: Actually, @ryanbooz has some pointers in his PR to the *changes* document; I think I'll highlight the breaking changes after I merge the two (i.e., in a later PR).

**Review by ryanbooz — APPROVED** [review]

**Pull request merged**